Assets/Scripts/InfiniteMana.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/*********************************************************************************************
+ * Script is meant to be applied to a Infinite Mana Spawner
+ * Inherits the entity class in order for the object to appear to take damage
+ * When TakeDamage is called, 1 - 3 mana orbs will spawn which can be picked up by the player
+ * Spawner will always regen back to full health which makes it indestructible.
+*********************************************************************************************/
+public class InfiniteMana : Entity
+{
+    public GameObject manaOrb; //Set this to the Mana Orb prefab
+    private float maxHealth; 
+    private float spawnerCordsX; 
+    private float spawnerCordsY;
+    private static System.Random rand = new System.Random();
+
+    public override void TakeDamage(float baseDamage)
+    {
+        base.TakeDamage(baseDamage);
+
+        Vector2 orbPosition;
+        for (int i = 0; i < rand.Next(1, 4); ++i) //Generate 1 - 3 mana orbs
```

Comment:
instead of using System's Random using [Random.Range](https://docs.unity3d.com/ScriptReference/Random.Range.html) is a bit more convenient and has more features